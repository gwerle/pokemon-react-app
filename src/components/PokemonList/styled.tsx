import styled, { css } from "styled-components";

export const ListContainer = styled.ul`
  ${({ theme }) => css`
    display: flex;
    flex-wrap: wrap;
    list-style: none;
    justify-content: center;
    padding: 0;

    li {
      padding: ${theme.spacings.xsmall};
      min-width: 200px;
    }
  `}
`;
