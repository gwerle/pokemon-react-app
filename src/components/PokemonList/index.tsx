import { InfiniteData } from "@tanstack/react-query";

import { PokemonListI } from "@/types";

import PokemonCard from "../PokemonCard";
import * as S from "./styled";

type Props = {
  data:
    | InfiniteData<{
        response: PokemonListI[];
        nextPage: string;
      }>
    | undefined;
};

export default function PokemonList({ data }: Props) {
  return (
    <S.ListContainer>
      {data?.pages.map(group =>
        group.response.map(pokemon => (
          <PokemonCard key={pokemon.url} pokemon={pokemon} />
        ))
      )}
    </S.ListContainer>
  );
}
