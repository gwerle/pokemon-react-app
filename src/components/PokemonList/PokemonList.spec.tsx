import { POKEMON_GROUPING_MOCK } from "@/utils/__mocks__/apiReturnMocks";
import { renderWithTheme } from "@/utils/tests/renderWithTheme";

import PokemonList from ".";

describe("<PokemonList />", () => {
  it("Renders without error", () => {
    const { container } = renderWithTheme(
      <PokemonList data={POKEMON_GROUPING_MOCK}></PokemonList>
    );

    expect(container).toMatchSnapshot();
  });
});
