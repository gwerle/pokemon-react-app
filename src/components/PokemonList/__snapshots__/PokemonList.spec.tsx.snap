// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonList /> Renders without error 1`] = `
<div>
  <ul
    class="sc-hLseeU iJEiVw"
  >
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
    <li>
      <div
        class="sc-gueYoa emUOiO"
      >
        <div
          class="sc-beqWaB iTdBVO"
          data-testid="skeleton"
          height="150px"
          width="100%"
        />
        <div
          class="sc-dmqHEX hZHgck"
        >
          <div>
            <strong />
          </div>
          <div />
        </div>
      </div>
    </li>
  </ul>
</div>
`;
